src/Functions/reinterpretAs.cpp

```diff
@@ -176,18 +176,23 @@ class FunctionReinterpret : public IFunction
                     size_t size = offsets_from.size();
                     auto & vec_res = col_res->getData();
                     vec_res.resize_fill(size);
+                    size_t copy_size = 0;
+                    size_t offset_to = 0;
 
                     size_t offset = 0;
                     for (size_t i = 0; i < size; ++i)
                     {
+                        copy_size = std::min(static_cast<UInt64>(sizeof(ToFieldType)), offsets_from[i] - offset - 1);
+                        offset_to = sizeof(ToFieldType) > copy_size ? sizeof(ToFieldType) - copy_size : 0;
```

Comment:
Let's move this calculation into the `else` case (l. 191.)

Reply:
Updated